internal/resolution/variablesources/catalogdsource.go

```diff
@@ -44,12 +64,12 @@ func (es *CatalogdEntitySource) Filter(ctx context.Context, filter input.Predica
 	return resultSet, nil
 }
 
-func (es *CatalogdEntitySource) GroupBy(ctx context.Context, fn input.GroupByFunction) (input.EntityListMap, error) {
+func (es *CatalogdVariableSource) GroupBy(ctx context.Context, fn input.GroupByFunction) (map[string][]deppy.Variable, error) {
```

Comment:
I believe we can remove `Get`, `GroupBy` and `Iterate` easily because we do not use them.

`Filter` has to be replaced somehow. It has a convenient API of accepting reusable predicate functions.

We might want to have some sort of `filter` package to help us work with variables. But I think it is reasonable to start without it and see whether we actually need it (e.g. do we have any repeating patterns?)

Reply:
I've been playing a bit with filtering & separate structures for catalog data. Here is what I came up with so far: https://github.com/operator-framework/operator-controller/pull/393